include/SFML/Window/ContextSettings.hpp

```diff
@@ -76,6 +81,8 @@ struct ContextSettings
     unsigned int majorVersion;      ///< Major number of the context version to create
     unsigned int minorVersion;      ///< Minor number of the context version to create
     Uint32       attributeFlags;    ///< The attribute flags to create the context with
+    std::string  vendorName;        ///< Name of the vendor of the OpenGL implementation
+    std::string  rendererName;      ///< Name of the OpenGL renderer implementation
```

Comment:
To continue on what Bromeon said, I see this as an introspection fonction and as such should go along with functions to get the number of cpu and stuff like that. I feel it would be wiser to design this larger API first. I suggest therefore that we don't consider this feature for 2.3 in order not to stall its release.